### PR TITLE
feat: add mongodb bridge (e5.0)

### PR DESCRIPTION
**Pull request by thalesmg**

### CODE REVIEW

**Comment by thalesmg** [comment]
I found an issue with this: https://github.com/emqx/emqx/blob/cf5d01fa09561d98723ec8a604051883147d145e/apps/emqx_connector/src/emqx_connector_mongo.erl#L393

It allows a configuration like this to be accepted:

```conf
  servers = ["localhost:27017"]
```

But that doesn't parse well: 

```
servers => [{[<<"localhost:27011">>],27017}]
```

I've tried to define it as a union to keep the same behavior for strings:

```
hoconsc:union([binary(), list(emqx_schema:ip_port())])
```

But then swagger doesn't parse the type name very well:

```
2022-09-01T11:20:21.872944-03:00 [error] Error in process <0.2875.0> on node 'emqx@127.0.0.1' with exit value:, {{nocatch,{minirest_trails_api_spec_error,{error,#{module => emqx_ee_bridge_mongodb,msg => <<"Unsupported Type">>,type => "ip_port()]"}}}},[{emqx_dashboard_swagger,typename_to_spec,2,[
```

Still trying to figure out a way to fix it.

File: apps/emqx_bridge/src/schema/emqx_bridge_schema.erl [hunk 1]
```diff
@@ -41,12 +41,16 @@ api_schema(Method) ->
     hoconsc:union(Broker ++ EE).
 
 ee_api_schemas(Method) ->
+    %% must ensure the app is loaded before checking if fn is defined.
+    ensure_loaded(emqx_ee_bridge, emqx_ee_bridge),
```
Comment (zmstone) [inline apps/emqx_bridge/src/schema/emqx_bridge_schema.erl:45]: I guess this only happens in tests, maybe move the loading to test helper module.
so we do not expose yet another part of ee to ce

File: lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_mongodb_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+%%------------------------------------------------------------------------------
+%% Helper fns
+%%------------------------------------------------------------------------------
+
+mongo_config(MongoHost0, MongoPort0, rs) ->
+    MongoHost = list_to_binary(MongoHost0),
+    MongoPort = integer_to_binary(MongoPort0),
+    Servers = <<MongoHost/binary, ":", MongoPort/binary>>,
+    Name = atom_to_binary(?MODULE),
+    #{
```
Comment (zmstone) [inline lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_mongodb_SUITE.erl:126]: nit: maybe use hocon format config text and call the schema module to parse it into this map.

File: apps/emqx_bridge/src/schema/emqx_bridge_schema.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_bridge/src/schema/emqx_bridge_schema.erl:45]: I actually saw this problem in the build:

```
2022-09-02T09:38:23.464800-03:00 [error] failed_to_check_schema: emqx_ee_conf_schema
2022-09-02T09:38:23.465549-03:00 [error] #{kind => validation_error,path => "bridges",reason => unknown_fields,unknown => <<"mongodb_rs">>,unmatched => <<"mqtt,webhook">>}
ERROR: call_hocon_failed: -v -t 2022.09.02.09.38.23 -s emqx_ee_conf_schema -c /home/thales/dev/emqx/emqx/_build/emqx-enterprise/rel/emqx/etc/emqx.conf -d /home/thales/dev/emqx/emqx/_build/emqx-enterprise/rel/emqx/data/configs generate
```

File: lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_mongodb_SUITE.erl (hunk 2 above)
Comment (thalesmg) [reply lib-ee/emqx_ee_bridge/test/emqx_ee_bridge_mongodb_SUITE.erl]: :heavy_check_mark:  https://github.com/emqx/emqx/pull/8844/commits/f1048babd882b1b037fe6c3b417d0dc08bb703a3

**Review by terry-xiaoyu — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**